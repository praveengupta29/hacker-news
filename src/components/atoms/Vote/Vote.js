import React from 'react';
import PropTypes from 'prop-types';

import UnVote from '../UnVote';
import UpVote from '../UpVote';

const Vote = ({ voteFeed, objectId, voted }) => {
  return voted ? (
    <UnVote voteFeed={voteFeed} objectId={objectId} />
  ) : (
    <UpVote voteFeed={voteFeed} objectId={objectId} />
  );
};

Vote.propTypes = {
  objectId: PropTypes.string.isRequired,
  voteFeed: PropTypes.func.isRequired,
  voted: PropTypes.bool,
};

Vote.defaultProps = {
  voted: false,
};

export default Vote;
