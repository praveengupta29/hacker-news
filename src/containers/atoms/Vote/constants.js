export const VOTE_FEED_KEY = 'HACKER_NEWS/FEED_LIST/VOTE_FEED_KEY';
export const VOTE_FEED = 'HACKER_NEWS/FEED_LIST/VOTE_FEED';
