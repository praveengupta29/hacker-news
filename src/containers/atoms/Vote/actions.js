import { VOTE_FEED } from './constants';

export const voteFeed = data => ({
  type: VOTE_FEED,
  data,
});
