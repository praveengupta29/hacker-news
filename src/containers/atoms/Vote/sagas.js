/* eslint-disable require-yield */
import { takeLatest } from 'redux-saga/effects';

import { VOTE_FEED, VOTE_FEED_KEY } from './constants';
import { LocalStorageUtil } from '../../../utils/localStorage';

export function* saveUpVoteFeedSaga(action) {
  const {
    data: { objectId, voted },
  } = action;
  const storage = new LocalStorageUtil();

  if (voted) {
    storage.saveItemInArray(VOTE_FEED_KEY, objectId);
  } else {
    storage.removeItemFromArray(VOTE_FEED_KEY, objectId);
  }

  // Here, an API request can be made to `up vote` feed
}

export default function* upVoteFeedSaga() {
  yield takeLatest(VOTE_FEED, saveUpVoteFeedSaga);
}
