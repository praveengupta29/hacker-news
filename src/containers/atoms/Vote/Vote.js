import { connect } from 'react-redux';

import { voteFeed } from './actions';

import Vote from '../../../components/atoms/Vote/Vote';

export const mapDispatchToProps = dispatch => {
  return {
    voteFeed: data => {
      dispatch(voteFeed(data));
    },
  };
};

export default connect(
  null,
  mapDispatchToProps,
)(Vote);
