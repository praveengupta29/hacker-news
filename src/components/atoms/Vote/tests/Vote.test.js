import React from 'react';
import { shallow } from 'enzyme';
import Vote from '../Vote';

describe('UnVote component', () => {
  const props = {
    voteFeed: () => {},
    objectId: '6382672',
    voted: true,
  };

  it('should render correctly with `UpVote` and match snapshot', () => {
    const component = shallow(<Vote {...props} />);
    expect(component).toMatchSnapshot();
  });

  it('should render correctly with `UnVote` and match snapshot', () => {
    props.voted = false;

    const component = shallow(<Vote {...props} />);
    expect(component).toMatchSnapshot();
  });
});
