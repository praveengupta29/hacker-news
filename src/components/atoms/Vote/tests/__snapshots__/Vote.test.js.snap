// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UnVote component should render correctly with \`UnVote\` and match snapshot 1`] = `
<UpVote
  objectId="6382672"
  voteFeed={[Function]}
/>
`;

exports[`UnVote component should render correctly with \`UpVote\` and match snapshot 1`] = `
<UnVote
  objectId="6382672"
  voteFeed={[Function]}
/>
`;
